### ISSUE TITLE
Fix false positives for SCSS interpolation in no-unknown-animations

### ISSUE THREAD

**Issue by ConduciveMammal** [stylelint/stylelint#5532]
> Clearly describe the bug

I'm using SCSS interpolation with math to create an incremental animation. Stylelint is stating `Unexpected unknown animation name "#{0.2"`. The interpolation isn't used for the name, simply the animation delay, so the error's not actually picking up the correct part.

> Which rule, if any, is the bug related to?

`no-unknown-animation`

> What code is needed to reproduce the bug?

```scss
@for $i from 0 to 13 {

  &:nth-child(#{$i}) {
    animation: fadeIn 1.2s cubic-bezier(0.39, 0.575, 0.565, 1) #{0.2 * $i}s both;
  }
}

@keyframes fadeIn {

  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": ["stylelint-config-shopify", "stylelint-config-sass-guidelines"],
  "plugins": ["stylelint-prettier", "stylelint-selector-bem-pattern"],
  "rules": {
    "indentation": 2,
    "string-quotes": "double",
    "no-duplicate-selectors": true,
    "color-no-hex": null,
    "color-hex-case": "lower",
    "color-named": "never",
    "color-hex-length": null,
    "selector-combinator-space-after": "always",
    "selector-attribute-quotes": "always",
    "selector-attribute-operator-space-before": "never",
    "selector-attribute-operator-space-after": "never",
    "selector-attribute-brackets-space-inside": "never",
    "declaration-block-trailing-semicolon": "always",
    "declaration-colon-space-before": "never",
    "declaration-colon-space-after": "always",
    "number-leading-zero": "always",
    "function-url-quotes": "always",
    "font-weight-notation": "numeric",
    "font-family-name-quotes": "always-where-recommended",
    "comment-whitespace-inside": "always",
    "comment-empty-line-before": "always",
    "at-rule-no-vendor-prefix": true,
    "at-rule-disallowed-list": null,
    "rule-empty-line-before": "always",
    "selector-pseudo-element-colon-notation": "double",
    "selector-pseudo-class-parentheses-space-inside": "never",
    "media-feature-range-operator-space-before": "never",
    "media-feature-range-operator-space-after": "always",
    "media-feature-parentheses-space-inside": "never",
    "media-feature-colon-space-before": "never",
    "media-feature-colon-space-after": "always",
    "selector-no-qualifying-type": null,
    "scss/dollar-variable-pattern": "^[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
    "order/properties-alphabetical-order": null,
    "selector-class-pattern": null,
    "selector-max-id": 1,
    "max-nesting-depth": [3, {
      "ignore": ["pseudo-classes"]
    }],
    "plugin/selector-bem-pattern": {
      "preset": "bem",
      "componentName": "[A-Z]+",
      "componentSelectors": {
        "initial": "^\\.{componentName}(?:-[a-z]+)?$",
        "combined": "^\\.combined-{componentName}-[a-z]+$"
      },
      "utilitySelectors": "^\\.util-[a-z]+$"
    }
  }
}

```

> Which version of stylelint are you using?

NPM:`13.7.1`
VSCode extension: `v0.86.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Running via VSCode extension

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, SCSS interpolation with math

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
Unexpected unknown animation name "#{0.2" (no-unknown-animations)stylelintno-unknown-animations
```

**Comment by jeddy3** [comment]
@ConduciveMammal Thanks for the report and for using the template.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

[In the `findAnimationName` util](https://github.com/stylelint/stylelint/blob/v14/lib/utils/findAnimationName.js#L21), we'll need to use the [`isStandardSyntaxValue` util](https://github.com/stylelint/stylelint/blob/v14/lib/utils/isStandardSyntaxValue.js) to return an empty array early if the value is non-standard. The value parser is for standard CSS and incorrectly parses the interpolation, throwing off the rest of the logic in the util.

Please branch off `v14` rather than `master`.
